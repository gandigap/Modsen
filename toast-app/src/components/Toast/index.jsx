import PropTypes from 'prop-types'
import ToastPortal from '../ToastPortal'
import { getDefaultColors, getIcons, getPadding } from '../../utils'
import { TOAST_TYPES, TOAST_SIZES } from '../../constants'

import { useEffect } from 'react'

const Toast = (props) => {
  const {
    label,
    size = getPadding(TOAST_SIZES.small).padding,
    toastType = TOAST_TYPES.info,
    color = getDefaultColors(toastType).font,
    bgColor = getDefaultColors(toastType).background,
    handleClick,
  } = { ...props }

  const style = () => {
    return {
      position: 'absolute',
      top: '0',
      left: '0',
      color: `${color}`,
      backgroundColor: `${bgColor}`,
      padding: `${getPadding(size).padding}`,
      borderRadius: '4px',
    }
  }

  const icon = getIcons(toastType, color)
  /* 
  useEffect(() => {
    const interval = setInterval(() => {
      document.body.removeChild(document.getElementById('root__portal'))
    }, 3000)

    return () => {
      clearInterval(interval)
    }
  }, []) */

  return (
    <ToastPortal>
      <div className="toast_wrapper" style={style()}>
        <div>{icon}</div>
        <span onClick={handleClick}>{label}</span>
      </div>
    </ToastPortal>
  )
}

Toast.propTypes = {
  label: PropTypes.string,
  handleClick: PropTypes.func,
  color: PropTypes.string,
  bgColor: PropTypes.string,
  toastType: PropTypes.oneOf([
    TOAST_TYPES.info,
    TOAST_TYPES.warning,
    TOAST_TYPES.error,
    TOAST_TYPES.success,
  ]),
  size: PropTypes.oneOf([TOAST_SIZES.small, TOAST_SIZES.medium, TOAST_SIZES.big]),
}

export default Toast
