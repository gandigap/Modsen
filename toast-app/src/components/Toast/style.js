import styled from 'styled-components'

export const ToastContainer = styled.div`
  font-size: 14px;
  box-sizing: border-box;
  position: absolute;
  z-index: 999999;
`

export const TypeIconContainer = styled.div`
  width: 40px;
  height: 40px;
`

export const CloseIconContainer = styled.div`
  width: 20px;
  height: 20px;
`
