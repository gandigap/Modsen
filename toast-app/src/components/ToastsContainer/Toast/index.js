import React, { useState } from 'react'
import PropTypes from 'prop-types'

import { getDefaultColors, getIcons } from '@/utils'
import {
  TOAST_TYPES,
  TOAST_SIZES,
  TOAST_ANIMATIONS,
} from '@/constants'
import { CloseIcon } from '../../icons'

import {
  StyledTypeIcon,
  StyledCloseIcon,
  StyledToastContainer,
  StyledToastTitle,
  StyledToastContent,
  StyledToastText,
} from './style'
import ErrorBoundary from '../../ErrorBoundary'

const Toast = (props) => {
  const [viewState, setViewState] = useState(true)

  const {
    id = '1',
    title,
    content,
    size,
    animationType,
    toastType = TOAST_TYPES.info,
    color,
    bgcolor,
    delay = 1000,
    handleClick,
    changeList,
  } = { ...props }

  const deleteToast = () => {
    setViewState(false)
    changeList(id)
    setTimeout(() => {
      if (handleClick) {
        handleClick(id)
      } else {
        document.getElementById(id).remove()
      }
    }, delay)
  }

  const typeIcon = getIcons(toastType, color)

  return (
    <ErrorBoundary>
      <StyledToastContainer
        id={id}
        className={
          viewState ? 'animation-start' : 'animation-end'
        }
        color={
          color === ''
            ? getDefaultColors(toastType).font
            : color
        }
        backgroundColor={
          bgcolor === ''
            ? getDefaultColors(toastType).background
            : bgcolor
        }
        animation={animationType}
        size={size}
        delay={delay}>
        <StyledTypeIcon>{typeIcon}</StyledTypeIcon>
        <StyledToastText>
          <StyledToastTitle>{title}</StyledToastTitle>
          <StyledToastContent>{content}</StyledToastContent>
        </StyledToastText>
        <StyledCloseIcon onClick={deleteToast} data-id={id}>
          <CloseIcon color={color} />
        </StyledCloseIcon>
      </StyledToastContainer>
    </ErrorBoundary>
  )
}

Toast.propTypes = {
  title: PropTypes.string,
  handleClick: PropTypes.func,
  animationType: PropTypes.oneOf([
    TOAST_ANIMATIONS.horizontal,
    TOAST_ANIMATIONS.vertical,
  ]),
  color: PropTypes.string,
  bgcolor: PropTypes.string,
  toastType: PropTypes.oneOf([
    TOAST_TYPES.info,
    TOAST_TYPES.warning,
    TOAST_TYPES.error,
    TOAST_TYPES.success,
  ]),
  size: PropTypes.oneOf([
    TOAST_SIZES.small,
    TOAST_SIZES.medium,
    TOAST_SIZES.big,
  ]),
  delay: PropTypes.number,
}

export default Toast
