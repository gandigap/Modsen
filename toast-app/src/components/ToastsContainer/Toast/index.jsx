import React from 'react'
import PropTypes from 'prop-types'

import {
  getDefaultColors,
  getIcons,
  getPadding,
} from '../../../utils'
import {
  TOAST_TYPES,
  TOAST_SIZES,
  TOAST_ANIMATIONS,
} from '../../../constants'
import { CloseIcon } from '../../icons'

import {
  StypedTypeIcon,
  StyledCloseIcon,
  StyledToastContainer,
  StyledToastTitle,
} from './style'
import ErrorBoundary from '../../ErrorBoundary'

const Toast = (props) => {
  const {
    id = '1',
    label,
    size = getPadding(TOAST_SIZES.small).padding,
    animationType = TOAST_ANIMATIONS.horizontal,
    toastType = TOAST_TYPES.info,
    color = getDefaultColors(toastType).font,
    bgColor = getDefaultColors(toastType).background,
    handleClick,
  } = { ...props }  

  const typeIcon = getIcons(toastType, color)  

  return (
    <ErrorBoundary>
      <StyledToastContainer
        id={id}
        className={'test'}        
        color={color}
        backgroundColor={bgColor}
        animation={animationType}
        size={size}>
        <StypedTypeIcon>{typeIcon}</StypedTypeIcon>
        <StyledToastTitle >
          {label}
        </StyledToastTitle>
        <StyledCloseIcon onClick={handleClick(id)} data-id={id}>
          <CloseIcon color={color} />
        </StyledCloseIcon>
      </StyledToastContainer>
    </ErrorBoundary>
  )
}

Toast.propTypes = {
  label: PropTypes.string,
  handleClick: PropTypes.func,
  animationType: PropTypes.oneOf([
    TOAST_ANIMATIONS.horizontal,
    TOAST_ANIMATIONS.vertical,
  ]),
  color: PropTypes.string,
  bgColor: PropTypes.string,
  toastType: PropTypes.oneOf([
    TOAST_TYPES.info,
    TOAST_TYPES.warning,
    TOAST_TYPES.error,
    TOAST_TYPES.success,
  ]),
  size: PropTypes.oneOf([
    TOAST_SIZES.small,
    TOAST_SIZES.medium,
    TOAST_SIZES.big,
  ]),
}

export default Toast
