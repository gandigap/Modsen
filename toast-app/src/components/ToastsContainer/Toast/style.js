import styled from 'styled-components'
import { TOAST_ANIMATIONS } from '@/constants'
import { getPadding } from '@/utils'
export const StyledToastContainer = styled.div`
  margin: 10px;
  padding: ${({ size }) => getPadding(size).padding};
  min-width: 200px;
  min-height: 40px;
  display: inline-flex;
  justify-content: space-between;
  align-items: center;
  border-radius: 10px;
  color: ${({ color }) => color};
  background-color: ${({ backgroundColor }) =>
    backgroundColor};
  animation: 0.5s
    ${(props) =>
      props.animation === TOAST_ANIMATIONS.vertical
        ? 'start-y'
        : 'start-x'}
    1s;

  &.animation-start {
    animation: 0.5s
      ${({ animation }) =>
        animation === TOAST_ANIMATIONS.vertical
          ? 'start-y'
          : 'start-x'}
      1s;
  }

  &.animation-end {
    animation: 0.5s
      ${({ animation }) =>
        animation === TOAST_ANIMATIONS.ease
          ? 'end-y'
          : 'end-x'}
      1s;
  }

  &:hover {
    -webkit-box-shadow: 4px 4px 8px 0px
      rgba(34, 60, 80, 0.2);
    -moz-box-shadow: 4px 4px 8px 0px rgba(34, 60, 80, 0.2);
    box-shadow: 4px 4px 8px 0px rgba(34, 60, 80, 0.2);
  }

  @keyframes start-y {
    from {
      transform: translateY(-100%);
      opacity: 0;
    }
    to {
      transform: translateY(0);
      opacity: 1;
    }
  }

  @keyframes end-y {
    from {
      transform: translateY(-100%);
      opacity: 0;
    }
    to {
      transform: translateY(0);
      opacity: 1;
    }
  }

  @keyframes start-x {
    from {
      transform: translateX(-100%);
      opacity: 0;
    }
    to {
      transform: translateX(0);
      opacity: 1;
    }
  }

  @keyframes end-x {
    from {
      transform: translateX(-100%);
      opacity: 0;
    }
    to {
      transform: translateX(0);
      opacity: 1;
    }
  }
`

export const StyledToastTitle = styled.span`
  flex-grow: 1;
  padding: 4px;
  font-size: 20px;
  font-weight: bold;
`

export const StypedTypeIcon = styled.div`
  width: 30px;
  height: 30px;
`

export const StyledCloseIcon = styled.button`
  margin: 0;
  padding: 0;
  width: 20px;
  height: 20px;
  display: inline-block;
  border: none;
  text-decoration: none;
  background: transparent;
  align-self: start;
  cursor: pointer;
`
