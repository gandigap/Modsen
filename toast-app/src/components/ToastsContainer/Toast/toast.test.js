/* eslint-disable no-undef */
import React from 'react'
import { shallow } from 'enzyme'
import Toast from './index.js'

describe('Toast', () => {
  it('should render Toast without props', () => {
    const component = shallow(<Toast />)
    expect(component).toMatchSnapshot()
  })
})
