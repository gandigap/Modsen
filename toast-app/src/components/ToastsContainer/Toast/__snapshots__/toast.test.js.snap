// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toast should render Toast without props 1`] = `
<ErrorBoundary>
  <styled.div
    className="animation-start"
    delay={2000}
    id="1"
  >
    <styled.div>
      <InfoIcon />
    </styled.div>
    <styled.div>
      <styled.h4 />
      <styled.p />
    </styled.div>
    <styled.button
      data-id="1"
      onClick={[Function]}
    >
      <CloseIcon />
    </styled.button>
  </styled.div>
</ErrorBoundary>
`;
